javascript/src/methods.js

```diff
@@ -543,6 +543,43 @@ methods.addGeoJSON = function(data, layerId, group, style) {
   if (typeof(data) === "string") {
     data = JSON.parse(data);
   }
+  let popupProperty = style.popupProperty;
+  let labelProperty = style.labelProperty;
+  delete style.popupProperty;
+  delete style.labelProperty;
+  let popupOptions = style.popupOptions;
+  let labelOptions = style.labelOptions;
+  delete style.popupOptions;
+  delete style.labelOptions;
```

Comment:
Better to pass these values as a separate object rather than piggyback them onto style and then pull them off. Same with highlightStyle below.